File: Overoom.React/src/services/room-hub/events/player/update-viewer-player.event.ts
/** Модель данных для события паузы */
export default interface UpdateViewerPlayerEvent {
  /** Идентификатор комнаты */
  id: string;

  /** Флаг паузы */
  onPause: boolean | null;

  /** Флаг полноэкранного режима */
  fullScreen: boolean | null;

  /** Позиция на временной шкале */
  timeLine: number | null;

  /** Скорость воспроизведения */
  speed: number | null;

  /** Номер сезона */
  season: number | null;

  /** Номер эпизода */
  episode: number | null;

  /** Список обновленных полей */
  updatedFields: string[];
}

File: Rooms.Application.Abstractions/RoomEvents/Player/UpdateViewerPlayerEvent.cs
namespace Rooms.Application.Abstractions.RoomEvents.Player;

/// <summary>
/// Модель данных для события обновления параметров плеера зрителя
/// </summary>
public class UpdateViewerPlayerEvent : RoomBaseEvent
{
    /// <summary>Идентификатор зрителя</summary>
    public required Guid Id { get; init; }

    /// <summary>Флаг паузы</summary>
    public bool? OnPause { get; set; }

    /// <summary>Флаг полноэкранного режима</summary>
    public bool? FullScreen { get; set; }

    /// <summary>Позиция на временной шкале</summary>
    public long? TimeLine { get; set; }
    
    /// <summary>Скорость воспроизведения</summary>
    public double? Speed { get; set; }

    /// <summary>Номер сезона</summary>
    public int? Season { get; set; }

    /// <summary>Номер эпизода</summary>
    public int? Episode { get; set; }
    
    /// <summary>Список обновленных полей</summary>
    public required IReadOnlyList<string> UpdatedFields { get; init; }
}